src/main/java/org/broadinstitute/hellbender/tools/walkers/groundtruth/GroundTruthScorer.java

```diff
@@ -0,0 +1,796 @@
+import org.broadinstitute.hellbender.utils.report.GATKReportTable;
+
+import java.io.*;
+import java.text.DecimalFormat;
+import java.util.*;
+import java.util.zip.GZIPOutputStream;
+
+@CommandLineProgramProperties(
+        summary = "Ground Truth Scorer",
+        oneLineSummary = "Score reads against a reference/ground truth",
+        programGroup = FlowBasedProgramGroup.class
+)
+@DocumentedFeature
+@ExperimentalFeature
+public class GroundTruthScorer extends ReadWalker {
+    private static final Logger logger = LogManager.getLogger(GroundTruthScorer.class);
+
+    public static final String OUTPUT_CSV_LONG_NAME = "output-csv";
+    public static final String REPORT_FILE_LONG_NAME = "report-file";
+    public static final String USE_SOFTCLIPPED_BASES_LONG_NAME = "use-softclipped-bases";
+    public static final String GENOME_PRIOR_LONG_NAME = "genome-prior";
+    public static final String FEATURES_FILE_LONG_NAME = "features-file";
+    public static final String NORMALIZED_SCORE_THRESHOLD_LONG_NAME = "normalized-score-threshold";
+    public static final String ADD_MEAN_CALL_LONG_NAME = "add-mean-call";
+    public static final String GT_NO_OUTPUT_LONG_NAME = "gt-no-output";
+    public static final String OMIT_ZEROS_FROM_REPORT = "omit-zeros-from-report";
+    public static final String QUALITY_PERCENTILES = "quality-percentiles";
+    public static final String EXCLUDE_ZERO_FLOWS = "exclude-zero-flows";
+
+    private static final int QUAL_VALUE_MAX = 60;
+    private static final int HMER_VALUE_MAX = 100; //TODO: This should become a parameter
+    private static final int BASE_VALUE_MAX = FlowBasedRead.DEFAULT_FLOW_ORDER.length() - 1;
+
+    private static final double NORMALIZED_SCORE_THRESHOLD_DEFAULT = -0.1;
+
+    private static class BooleanAccumulator {
+        long falseCount;
+        long trueCount;
+        BooleanAccumulator[] bins;
+
+        void add(final boolean b) {
+            if (b) {
+                trueCount++;
+            } else {
+                falseCount++;
+            }
+        }
+
+        void add(final boolean b, final int bin) {
+            add(b);
+            if ( bins != null && bin >= 0 && bin < bins.length ) {
+                bins[bin].add(b);
+            } else {
+                logger.warn("bin out of range; " + bin + ", range: [0," + bins.length + "), clipped");
+                bins[Math.max(0, Math.min(bin, bins.length - 1))].add(b);
+            }
+        }
+        void add(final boolean b, final int bin, final int bin2) {
+            add(b);
+            if ( bins != null && bin >= 0 && bin < bins.length ) {
+                bins[bin].add(b, bin2);
+            } else {
+                logger.warn("bin out of range; " + bin + ", range: [0," + bins.length + "), clipped");
+                bins[Math.max(0, Math.min(bin, bins.length - 1))].add(b, bin2);
+            }
+        }
+        void add(final boolean b, final int bin, final int bin2, final int bin3) {
```

Comment:
I found this accumulator a bit confusing, could you add a brief comment for the class or these functions?